sopel/coretasks.py

```diff
@@ -762,18 +763,28 @@ def acct_warn(bot, cap):
 
 def receive_cap_ack_sasl(bot):
     # Presumably we're only here if we said we actually *want* sasl, but still
-    # check anyway.
-    password = None
-    mech = None
-    if bot.config.core.auth_method == 'sasl':
-        password = bot.config.core.auth_password
-        mech = bot.config.core.auth_target
-    elif bot.config.core.server_auth_method == 'sasl':
-        password = bot.config.core.server_auth_password
-        mech = bot.config.core.server_auth_sasl_mech
+    # check anyway in case the server glitched.
+    password, mech = _get_sasl_pass_and_mech(bot)
```

Comment:
OK it's kind of a nitpick here: if all you need is the settings, then just provide the settings. I've added other suggestion to reflect that change, so you can commit them in one go.

```suggestion
    password, mech = _get_sasl_pass_and_mech(bot.config)
```